app/controllers/MrnEntryController.scala

```diff
@@ -59,33 +56,19 @@ class MrnEntryController @Inject() (
       .fold(errorForm => Future.successful(BadRequest(mrnEntry(errorForm))), updateUserAnswersAndRedirect(_, req.userAnswers))
   }
 
-  def autoFill(mrn: String, maybeRefererUrl: Option[RedirectUrl] = None): Action[AnyContent] =
+  def autoFill(mrn: String): Action[AnyContent] =
     (authenticate andThen verifiedEmail andThen getData).async { implicit req =>
-      val updatedAnswers = decodeRefererUrl(maybeRefererUrl)
-        .map(decodedBackLink => req.userAnswers.copy(mrnPageRefererUrl = Some(decodedBackLink)))
-        .getOrElse(req.userAnswers)
-
-      MRN(mrn)
-        .map(updateUserAnswersAndRedirect(_, updatedAnswers))
```

Comment:
You are no-longer calling `updateUserAnswersAndRedirect` so the `mrn` value is not being saved in the cache any more.

We don't need the `mrnPageRefererUrl` saved anymore but we certainly do still need the `mrn` that is passed in saved to the cache!